### ISSUE TITLE
Typescript implicit any from RequestUrlTemplate

### ISSUE THREAD

**Issue by bakesteve** [NSwag/NSwag#563]
in typescript we prevent the use of implicit any
https://github.com/NSwag/NSwag/blob/master/src/NSwag.CodeGeneration.TypeScript/Templates/RequestUrlTemplate.tt#L42 generates script that wont parse with this rule.
Is there any way round this? or a way to override a specific template ?
If not, would you consider a PR on this?

**Comment by rsuter** [comment]
Yes we should fix that. Where exactly is the problem? Yes, a pr is welcome

**Comment by rsuter** [comment]
How would you correctly type that?

**Comment by bakesteve** [comment]
its a little tricky, typescript isnt massively happy with for .. in
I'll try a couple of tweaks and add them here / via a PR

**Comment by rsuter** [comment]
@grovesNL is this issue fixed?

**Comment by grovesNL** [comment]
@rsuter Not yet. It wasn't caught in [my branch's changes](https://github.com/NSwag/NSwag/compare/master...grovesNL:typescript-type-improvements) yet either because we don't seem to have any sample TypeScript clients that use that type (see https://github.com/NSwag/NSwag/search?utf8=%E2%9C%93&q=for+let+attr+in+item&type=Code). We should add a client method that uses an object array so I can correct this in my branch.

**Comment by rsuter** [comment]
I think there is no way that nswag generates this swagger spec... It must be manually defined. I also dont have this in production...

**Comment by grovesNL** [comment]
@rsuter Is it supported by the Swagger spec, or should we manually define it for testing?

**Comment by rsuter** [comment]
@bakesteve can you provide a sample swagger spec?

**Comment by MohammadRoshani** [comment]
https://github.com/RicoSuter/NSwag/issues/1590#issuecomment-1296215179
